library/ssl_tls13_keys.c

```diff
@@ -1407,17 +1438,48 @@ int mbedtls_ssl_tls13_generate_application_keys(
     size_t hash_len;
 
     /* Variables relating to the cipher for the chosen ciphersuite. */
```

Comment:
Partly preexisting: there's some code to retrieve cipher suite information in `mbedtls_ssl_tls13_generate_application_keys` and `mbedtls_ssl_tls13_generate_handshake_keys`, and this patch substantially increases the length of that code. Can it be unified into a single auxiliary function?